component/src/main/ballerina/ballerina/net/reliable/httpr/httpr.bal

```diff
@@ -0,0 +1,373 @@
+    action patch (string path, http:Request req) (http:Response, http:HttpConnectorError) {
+        blob storableStream = convertToBlob(req, connectorOptions, serviceUri, path, "patch") ;
+        return storeMessage(storableStream, guaranteedProcessor);
+    }
+
+    @Description { value:"DELETE action implementation of the reliable HTTP Connector"}
+    @Param { value:"path: Request path" }
+    @Param { value:"req: An HTTP Request struct" }
+    @Return { value:"The response message indicates safe persistance of the incoming request" }
+    @Return { value:"Error occured during HTTP reliable client storage action" }
+    action delete (string path, http:Request req) (http:Response, http:HttpConnectorError) {
+        blob storableStream = convertToBlob(req, connectorOptions, serviceUri, path, "delete") ;
+        return storeMessage(storableStream, guaranteedProcessor);
+    }
+
+    @Description { value:"OPTIONS action implementation of the reliable HTTP Connector"}
+    @Param { value:"path: Request path" }
+    @Param { value:"req: An HTTP Request struct" }
+    @Return { value:"The response message indicates safe persistance of the incoming request" }
+    @Return { value:"Error occured during HTTP reliable client storage action" }
+    action options (string path, http:Request req) (http:Response, http:HttpConnectorError) {
+        blob storableStream = convertToBlob(req, connectorOptions, serviceUri, path, "options") ;
+        return storeMessage(storableStream, guaranteedProcessor);
+    }
+
+    @Description { value:"forward action can be used to invoke an HTTP call with incoming request HTTPVerb"}
+    @Param { value:"path: Request path" }
+    @Param { value:"req: An HTTP Request struct" }
+    @Return { value:"The response message indicates safe persistance of the incoming request" }
+    @Return { value:"Error occured during HTTP reliable client storage action" }
+    action forward (string path, http:Request req) (http:Response, http:HttpConnectorError) {
+        blob storableStream = convertToBlob(req, connectorOptions, serviceUri, path, "forward") ;
+        return storeMessage(storableStream, guaranteedProcessor);
+    }
+
+    @Description { value:"Invokes an HTTP call with the specified HTTP verb."}
+    @Param { value:"HTTPVerb: HTTP verb value" }
+    @Param { value:"path: Resource path " }
+    @Param { value:"req: An HTTP Request struct" }
+    @Return { value:"The response message indicates safe persistance of the incoming request" }
+    @Return { value:"Error occured during HTTP reliable client storage action" }
+    action execute (string HTTPVerb, string path, http:Request req) (http:Response, http:HttpConnectorError) {
+        blob storableStream = convertToBlob(req, connectorOptions, serviceUri, path, HTTPVerb) ;
+        return storeMessage(storableStream, guaranteedProcessor);
+    }
+}
+
+public connector HttpClient (string serviceUri, http:Options connectorOptions) {
```

Comment:
shall we make this private for the time being? (since it is not in a usable state yet) and I'm +1 for changing this connector name :) IMO developer should get the feeling he is using a reliable client. Since this has the same name as http client, he may get confused.

Reply:
Updated the access modifier.
Purpose of this client is to work with embedded MB4. But its not there at the moment, therefore it will not be used. If we are giving a different name to the client, I think we have to have some kind of a pattern to go ahead. Because in future, there can be Guaranteed clients for other transports as well